katdal/chunkstore_s3.py

```diff
@@ -598,6 +607,19 @@ def complete_request(self, method, url, chunk_name='',
             else:
                 return result
 
+    def _verify_bucket(self, url, chunk_error=None):
+        """Check that bucket associated with `url` exists and is not empty."""
+        bucket = _bucket_url(url)
+        try:
+            response = self.complete_request('GET', bucket, process=lambda r: r)
```

Comment:
If you pass `max-keys=1` as a query parameter it'll limit the size of the response when the bucket is non-empty.

Reply:
Good point, done.